src/main/frontend/handler/route.cljs

```diff
@@ -43,21 +45,38 @@
   []
   (redirect! {:to :whiteboards}))
 
+;; Named block links only works on web (and publishing)
+(if util/web-platform?
+  (defn- default-page-route [page-name-or-block-uuid]
+    ;; Only query if in a block context
+    (let [block (when (uuid? page-name-or-block-uuid)
+                  (model/get-block-by-uuid page-name-or-block-uuid))]
+      (if (get-in block [:block/properties :heading])
+        {:to :page-block
+         :path-params {:name (get-in block [:block/page :block/name])
+                       :block-route-name (model/heading-content->route-name (:block/content block))}}
+        {:to :page
+         :path-params {:name (str page-name-or-block-uuid)}})))
+
+  (defn- default-page-route [page-name]
+    {:to :page
+     :path-params {:name (str page-name)}}))
+
 (defn redirect-to-page!
```

Comment:
I confirmed this fn handles all block level links so by adding named block support here, it just works for any feature e.g. zoom or search for collapsed block